samcli/lib/config/samconfig.py

```diff
@@ -102,20 +112,15 @@ def put(self, cmd_names, section, key, value, env=DEFAULT_ENV):
         key : str
             Key to write the data under
         value : Any
-            Value to write. Could be any of the supported TOML types.
+            Value to write. Could be any of the supported types.
         env : str
             Optional, Name of the environment
 
         Raises
         ------
-        tomlkit.exceptions.TOMLKitError
+        SamConfigFileReadException
             If the data is invalid
         """
-
-        if self.document is None:
-            # Checking for None here since a TOMLDocument can include a
-            # 'body' property but still be falsy without a 'value' property
-            self._read()
```

Comment:
Asking for my understanding, why is this `_read` method call removed? If this method call needs to be removed, is there anything else that raises `SamConfigFileReadException` exception as mentioned in line 121?

Reply:
Formerly, the `_read` method was being called in every function that used `self.document`, but now it performs the read once upon initialization, so we don't have to scatter function calls like this any time we want to look through the config.

Reply:
Understood :+1: , can we update the doc-string if `SamConfigFileReadException` is not raised in this method?